seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/org/apache/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java

```diff
@@ -68,6 +68,10 @@ public Configuration provide(Configuration originalConfiguration) {
                     // 1. Get config value from the system property
                     result = originalConfiguration.getConfigFromSys(rawDataId);
 
+                    if (result == null) {
+                        result = originalConfiguration.getConfig(rawDataId);
+                    }
```

Comment:
Why is this code being added?

Reply:
From an overall perspective, originalConfiguration.getConfig(rawDataId); is used to leverage getLatestConfig to fetch the configuration for rawDataId. After analysis, it indeed functionally covers getConfigFromSys. Therefore, it seems more appropriate to replace:
```java
// 1. Get config value from the system property 
result = originalConfiguration.getConfigFromSys(rawDataId);
```
with:
```java
// 1. Get config value from the system property 
result = originalConfiguration.getConfig(rawDataId);
```
instead of adding:
```java
if (result == null) { 
  result = originalConfiguration.getConfig(rawDataId); 
}
```

Reply:
Then why still keep `originalConfiguration.getConfigFromSys(rawDataId);`?

Reply:
你目前的代码破坏了原本代码的目的，这段代码是为了先读环境变量和-D参数，然后再尝试读取spring中的配置，而你目前的改动变成了先读环境变量，在读配置中心，再读spring了，而这个SpringBootConfigurationProvider只是为了读取基本配置，配置中心中的配置有CURRENT_FILE_INSTANCE进行读取，不需要通过`ORIGIN_FILE_INSTANCE_REGISTRY` 也就是SpringBootConfigurationProvider读取
Your current code breaks the original purpose. This section of the code is intended to first read environment variables and -D parameters, and then attempt to read the configuration from Spring. However, with your modification, it now first reads from environment variables, then from the configuration center, and finally from Spring. The SpringBootConfigurationProvider is only meant to read basic configurations, and configurations in the configuration center are read via CURRENT_FILE_INSTANCE, so there’s no need to use ORIGIN_FILE_INSTANCE_REGISTRY, i.e., the SpringBootConfigurationProvider, to read them.

Reply:
Use SeataApplicationListener to set springConfigurableEnvironment,ensuring it's available during configuration retrieval.
This avoids the need for additional null checks and ensures proper loading timing.